### ISSUE TITLE
proper cluster shutdown

### ISSUE THREAD

**Issue by dkhokhlov** [dask/distributed#3581]
What is correct way to shutdown the cluster (processes=True) without getting intermittent harmless error messages like this:

```
Traceback (most recent call last):
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/multiprocessing/queues.py", line 242, in _feed
    send_bytes(obj)
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/multiprocessing/connection.py", line 200, in send_bytes
    self._send_bytes(m[offset:offset + size])
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/multiprocessing/connection.py", line 404, in _send_bytes
    self._send(header + buf)
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/multiprocessing/connection.py", line 368, in _send
    n = write(self._handle, buf)
BrokenPipeError: [Errno 32] Broken pipe

```

Cluster start code:

```
    cluster = LocalCluster(n_workers=n_workers, processes=True, threads_per_worker=1, local_directory=workspace_path)
    client = Client(cluster)

```
Cluster shutdown code:

```
    client.close()

```

Also It looks like dask/nanny is trying to restart workers in application exit hooks ...

**Comment by mrocklin** [comment]
I wouldn't expect to see that line.  If you're seeing it then perhaps it is a bug.

**Comment by dkhokhlov** [comment]
The workaround I found so far:

```
    # silence chatty worker's nanny
    nanny_logger = logging.getLogger('distributed.nanny')
    nanny_logger.setLevel(logging.ERROR)
    client.shutdown()

```

**Comment by dkhokhlov** [comment]
Doing client.shutdown() instead of client.close() causes another `intermittent` error messages:

```
Traceback (most recent call last):
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/tornado/ioloop.py", line 743, in _run_callback
    ret = callback()
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/tornado/ioloop.py", line 767, in _discard_future_result
    future.result()
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/worker.py", line 918, in heartbeat
    raise e
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/worker.py", line 891, in heartbeat
    metrics=await self.get_metrics(),
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/utils_comm.py", line 391, in retry_operation
    operation=operation,
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/utils_comm.py", line 379, in retry
    return await coro()
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/core.py", line 757, in send_recv_from_rpc
    result = await send_recv(comm=comm, op=key, **kwargs)
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/core.py", line 540, in send_recv
    response = await comm.read(deserializers=deserializers)
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/comm/tcp.py", line 208, in read
    convert_stream_closed_error(self, e)
  File "/home/dmitri.khokhlov/.conda/envs/train/lib/python3.7/site-packages/distributed/comm/tcp.py", line 123, in convert_stream_closed_error
    raise CommClosedError("in %s: %s" % (obj, exc))
distributed.comm.core.CommClosedError: in <closed TCP>: Stream is closed

```

Doing nothing - not calling close() and shutdown() - the only workaround I found so far which does not  produce error messages.

**Comment by jrbourbeau** [comment]
@dkhokhlov what version of `distributed` are you using? The changes in #3529 were included in the 2.12.0 release and should impact the `Stream is closed` message you're seeing

**Comment by dkhokhlov** [comment]
That was 2.10. 
2.13 works.
